Remove CoreCLR TargetSpecific test build jobs

sdmaclea opened the pull request:
Refactor targetSpecific test build
Remove targetSpecific build-test-jobs.yml jobs
Build targetSpecific tests in run-test-job.yml jobs

sdmaclea committed:
Remove CoreCLR TargetSpecific test build jobs

Refactor targetSpecific test build
Remove targetSpecific build-test-jobs.yml jobs
Build targetSpecific tests in run-test-job.yml jobs

sdmaclea commented on eng/pipelines/common/templates/runtimes/run-test-job.yml:
```diff
@@ -236,6 +238,24 @@ jobs:
+        # Build targetSpecific managed test components
+        - script: $(coreClrRepoRootDir)build-test$(scriptExt) targetSpecific skipnative skipgeneratelayout skiptestwrappers $(buildConfig) $(archType) $(crossArg) $(priorityArg) ci $(librariesOverrideArg)
+          displayName: Build managed test components
+
```
The earliest draft of this was also adding an archive of the tests.
I was fiddling with two variants.  
1. Archive all tests
2. Archive target specific tests

I though that if performance was really critical, no artifact would be faster.

sdmaclea commented:
One potential other problem with this approach is when we run multiple jobs on the same tests.  R2R & !R2R & JitStress* &&&&....

Every job would be building the target specific tests.

sdmaclea commented:
```
Artifact MicrosoftNetSdkIlPackage_*_checked not found for build
```

I guess the target specific build was generating this too...

sdmaclea committed:
Fix Microsoft.NET.Sdk.IL package download

sdmaclea commented:
This seems to be working now.
I am going to push a change to the display names, but this could use at least a preliminary review.

sdmaclea committed:
Adjust display names

hoyosjs commented:
The other disadvantage here is retrying a test run will cause a rebuild too. Not sure if this matters too much.

safern commented:
> The other disadvantage here is retrying a test run will cause a rebuild too. Not sure if this matters too much.

While that is true, I think this will bring faster builds, less legs meaning less prone to infra errors, and less times hitting the network to download artifacts, etc. So as long as the tests are reliable enough, we shouldn't be specializing for retry.

sdmaclea commented:
There are 5 readytorun tests which are not happy with this change.  I am assuming it is because the tests were `crossgen2` for `x64`.

I assume therefore, these tests need to be marked `ArchSpecific`

trylek commented on eng/pipelines/common/templates/runtimes/run-test-job.yml:
```diff
@@ -236,15 +238,35 @@ jobs:
 
+
+    # Build target specific tests
+    - ${{ if ne(parameters.corefxTests, true) }}:
```
Nit - you should be able to combine the two if statements using the and() function and remove one level of nesting ;-).

trylek reviewed (approved):
Looks great, thank you for all the improvements you're making to our pipelines!

sdmaclea committed:
Make 5 tests target specific

These tests are failing in PR runs when built on OSX and
run on specific platforms

sdmaclea committed:
Feedback

sdmaclea commented:
I marked the 5 tests failing in p0 target specific..

Running an outerloop p1 run here https://dev.azure.com/dnceng/public/_build/results?buildId=627124&view=results

I am expecting to have to mark some p1 tests too.

sdmaclea committed:
Fix disable of 5 tests

sdmaclea commented:
Closing in favor of #35783
